core/api/pod-api/src/fs.ts

```diff
@@ -30,7 +30,7 @@ export interface FS {
     readFile(path: string): Promise<Uint8Array>;
     writeFile(path: string, content: string, options: EncodingOptions): Promise<void>;
     stat(path: string): Promise<Stats>;
-    readdir(path: string): Promise<string[]>;
+    readdir(path: string): Promise<{ [key: string]: string }[]>;
```

Comment:
The problem is right here. Please remember this is simply an interface, and the interface needs to be implemented in practice to make it work. The test that's failing is in charge of seeing if that interface really holds. Let's step back for a minute to really ascertain what's that interface: this is simply the interface of a `fs`-like object.  Any class with that interface will be plug-in-compatible with fs, so this is one of the things you need to *not* change.
And that's because, in general, we will not be implementing any fs-like object; we'll be using any ready-made, off-the-shelf module that will fit that interface. In the error you're getting this code is involved:
https://github.com/polypoly-eu/polyPod/blob/6669f52938279565bfeb39ef5a286c41298c8b9c/core/api/pod-api/src/tests/default.test.ts#L5-L10
Please check the module that's used for the mock, `memfs`, which is plug-in compatible with `fs`, exactly like *our* fs should be. Let's get to the default implementation above: :arrow_up:

Reply:
No, our FS should *not* be plug-in compatible with Node.js' FS API, that's a design we are explicitly changing here. We have different challenges, e.g. working with both IDs and paths. Seems like a good time to just eliminate _memfs_, it caused a ton of trouble in the past when being used elsewhere. `BrowserPod` already has its own FS implementation (of our actual API).

`DefaultPod` is not something we're using anywhere, most implementations there just throw -> The build should be green, but we don't really need anything in `DefaultPod` to work.

Reply:
More on topic: Similarly to `stat`, I think this should return an object of a certain class, not an ad-hoc defined generic object :D

Reply:
Well, it was so far, and that allowed us to use some other fs-compatible modules like `memfs` for testing purposes. The fact that it's stopped being compatible has caused the test fail.

Reply:
It's already just accidentally compatible - we work with IDs everywhere, not paths. Considering how the default pod implementation using _memfs_ is unused, and the tests we currently have are therefore just testing _that_ memfs is being used in some unused code, we should just disable those for now, and get rid of them later. Adding some tests for _actually_ used code in the process!

As for naming suggestions for this result object class, I was thinking `File` could be simple and sweet.